pkg/virt-handler/container-disk/mount.go

```diff
@@ -106,25 +105,20 @@ func (m *mounter) deleteMountTargetRecord(vmi *v1.VirtualMachineInstance) error
 		return fmt.Errorf("unable to find container disk mounted directories for vmi without uid")
 	}
 
-	recordFile := filepath.Join(m.mountStateDir, string(vmi.UID))
-
-	exists, err := diskutils.FileExists(recordFile)
-	if err != nil {
-		return err
+	record := vmiMountTargetRecord{}
+	err := m.checkpointManager.Get(string(vmi.UID), &record)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("failed to get a checkpoint %s, %w", vmi.UID, err)
 	}
-
-	if exists {
-		record, err := m.getMountTargetRecord(vmi)
-		if err != nil {
-			return err
-		}
-
+	if !errors.Is(err, os.ErrNotExist) {
 		for _, target := range record.MountTargetEntries {
 			os.Remove(target.TargetFile)
```

Comment:
It's not related directly to this PR, but if we care about removing the file, we should probably check the return value of `os.Remove`.

Reply:
I agree but I want to keep it out of the PR.